pkg/channel/consolidated/reconciler/controller/kafkachannel.go

```diff
@@ -530,12 +590,25 @@ func (r *Reconciler) updateKafkaConfig(ctx context.Context, configMap *corev1.Co
 	// Eventually the previous config should be snapshotted to delete Kafka topics
 	r.kafkaConfig = kafkaConfig
 	r.kafkaConfigError = err
+	factory := func() (sarama.ClusterAdmin, error) {
+		return source.MakeAdminClient(controllerAgentName, r.kafkaAuthConfig, kafkaConfig.Brokers)
+	}
+	ac, err := kafka.NewAdminClient(ctx, factory)
```

Comment:
not sure I like the `kafka.NewAdminClient` name, and can't we just pass the functtion to the `NewAdminClient`?

Reply:
can you please elaborate more about not liking the name? naming is hard and sometimes it becomes a personal preference, unless I'm breaking some naming convention?

Reply:
I'd do:

```golang
	ac, err := kafka.NewAdminClient(ctx, func() (sarama.ClusterAdmin, error) {
		return source.MakeAdminClient(controllerAgentName, r.kafkaAuthConfig, kafkaConfig.Brokers)
	})
```

than I don't like the `kafka.NewAdminClient` name... but yeah... let's leave the name alone  but the above code, I'd inline :shrug: